core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -72,26 +72,32 @@ trait TestPrematureExit {
     mainObject.printStream = printStream
 
     @volatile var exitedCleanly = false
-    mainObject.exitFn = (_) => exitedCleanly = true
-
-    @volatile var exception: Exception = null
-    val thread = new Thread {
-      override def run() = try {
-        mainObject.main(input)
-      } catch {
-        // Capture the exception to check whether the exception contains searchString or not
-        case e: Exception => exception = e
-      }
+    def withFakeExit(body: => Unit): Unit = {
```

Comment:
If this is the only occurrence, why a helper method? it seems more complex.
Also don't you want a try-finally to ensure the exitFn is restored?

Reply:
It's ok to just modify/restore exitFn in a straight way, but use a helper method here will make the modification of exitFn more clear?
moreover, I was wondering if a try-finally block is necessary? there is already a try-catch block in the func body of the only occurrence.

Reply:
I agree that a try-finally block is necessary, because the func body throws exception.